ddl/ddl_worker.go

```diff
@@ -155,6 +155,12 @@ func (d *ddl) finishDDLJob(t *meta.Meta, job *model.Job) (err error) {
 		metrics.DDLWorkerHistogram.WithLabelValues(metrics.WorkerFinishDDLJob, metrics.RetLabel(err)).Observe(time.Since(startTime).Seconds())
 	}()
 	switch job.Type {
+	case model.ActionAddIndex:
+		if job.State != model.JobStateRollbackDone {
+			break
+		}
+		// ActionAddIndex needs to delete ranges when it needs to be rolled back.
+		fallthrough
```

Comment:
`fallthrough` is rarely used, it's hard to read.
How about extracting a function `shouldDeleteRange`?